ml-agents/mlagents/trainers/learn.py

```diff
@@ -268,26 +308,36 @@ def from_argparse(args: argparse.Namespace) -> "RunOptions":
           configs loaded from files.
         """
         argparse_args = vars(args)
+        full_config = {}
+        full_config.update(argparse_args)
+        # Since argparse accepts file paths in the config options which don't exist in CommandLineOptions,
+        # these keys will need to be deleted to use the **/splat operator below.
+        full_config.pop("trainer_config_path")
```

Comment:
Wonder if the path issue could be fixed upstream of this.  Maybe with a custom action..

Reply:
Not sure if it's what you had in mind, but I added custom action that removes the `trainer_config_path` attribute from the namespace and adds it to a class variable which can be accessed by `learn.py`. Also moved custom actions and the huge `create_parser` method to a separate file.